src/Ledger/ScriptValidation.agda

```diff
@@ -89,25 +76,21 @@ txInfo l pp utxo tx = record
   ; vkKey = reqSigHash
   } where open Tx tx; open TxBody body
 
-DelegateOrDeReg : DCert → Set
-DelegateOrDeReg = λ where
-  (delegate _ _ _ _) → ⊤
-  (dereg _)          → ⊤
-  (regpool _ _)      → ⊥
-  (retirepool _ _)   → ⊥
-  (regdrep _ _ _)    → ⊤
-  (deregdrep _)      → ⊤
-  (ccreghot _ _)     → ⊥
+data DelegateOrDeReg : DCert → Set where instance
+  delegate  : ∀ {x y z w} → DelegateOrDeReg (delegate x y z w)
+  dereg     : ∀ {x} →       DelegateOrDeReg (dereg x)
+  regdrep   : ∀ {x y z} →   DelegateOrDeReg (regdrep x y z)
+  deregdrep : ∀ {x} →       DelegateOrDeReg (deregdrep x)
```

Comment:
I changed this due to (what I believe is) an [Agda bug](https://github.com/agda/agda/issues/7040), which resulted in instance resolution failing for `⊤ ⁇` and `⊥ ⁇`.